packages/vite/src/node/ssr/ssrModuleLoader.ts

```diff
@@ -91,11 +94,12 @@ async function instantiateModule(
 
   const ssrImport = async (dep: string) => {
     if (dep[0] !== '.' && dep[0] !== '/') {
-      return nodeRequire(
+      return await nodeRequireOrImport(
         dep,
         mod.file,
         server.config.root,
-        !!server.config.resolve.preserveSymlinks
+        !!server.config.resolve.preserveSymlinks,
+        server,
```

Comment:
We could pass `server.config` and remove the `root` and `preserveSymlinks` arguments.